Dynamic webhooks

realshuting opened the pull request:
## Related issue

Closes https://github.com/kyverno/kyverno/issues/1981.
Closes https://github.com/kyverno/kyverno/issues/2424.
Closes https://github.com/kyverno/kyverno/issues/2321.
Closes https://github.com/kyverno/kyverno/issues/1995.
Closes https://github.com/kyverno/kyverno/issues/893.
Relates to #1983.

<!--
Please link the GitHub issue this pull request resolves in the format of `#1234`. If you discussed this change
with a maintainer, please mention her/him using the `@` syntax (e.g. `@JimBugwadia`).

If this change neither resolves an existing issue nor has sign-off from one of the maintainers, there is a
chance substantial changes will be requested or that the changes will be rejected.

You can discuss changes with maintainers in the [Kyverno Slack Channel](https://kubernetes.slack.com/).
-->

## Milestone of this PR
<!--

Add the milestone label by commenting `/milestone 1.2.3`.

-->
## What type of PR is this

<!--

> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading white spaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
-->

## Proposed Changes
Flags:
- `webhooktimeout`: default webhookTimeout is set to 10s, and it is updated to the longest timeout of the configured policies with the same `FailurePolicy`.
- "auto-update-webhooks": auto-update of the webhook configurations is enabled by default. Set it to false to disable it.

Behaviors:
- `resourceFilters`: the `resourceFilters` only filter out resources if auto-update is disabled. There's no need to configure it if auto-update is enabled (by default).
- `policy.status.ready`: once the matching kinds of a policy are updated to the webhook successfully, the policy is marked as ready to serve the admission requests.
- `policy.spec.failurePolicy`: default is mutated to `Fail`, `Ignore` has to be set manually.
- `policy.spec.webhookTimeoutSeconds`, default is set to 10s

<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. 

***NOTE***: If this PR results in new or altered behavior which is user facing, you **MUST** read and follow the steps outlined in the [PR documentation guide](pr_documentation.md) and add Proof Manifests as defined below.
-->

### Proof Manifests
See regression tests https://docs.google.com/document/d/1Y7_7ow4DgCLyCFQcFVz1vHclghazAKZyolIfprtNURc/edit#heading=h.ifbnt97ake0i.

<!--
Read and follow the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) for more details first. This section is for pasting your YAML manifests (Kubernetes resources and Kyverno policies) which allow maintainers to prove the intended functionality is achieved by your PR. Please use proper fenced code block formatting, for example:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: roles-dictionary
  namespace: default
data:
  allowed-roles: "[\"cluster-admin\", \"cluster-operator\", \"tenant-admin\"]"
```
-->

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [] I have added tests that prove my fix is effective or that my feature works.
- [] My PR contains new or altered behavior to Kyverno and
  - [] I have added or changed [the documentation](https://github.com/kyverno/website) myself in an existing PR and the link is:
  <!-- Uncomment to link to the PR -->
  <!-- https://github.com/kyverno/website/pull/123 -->
  - [] I have raised an issue in [kyverno/website](https://github.com/kyverno/website) to track the doc update and the link is:
  <!-- Uncomment to link to the issue -->
  <!-- https://github.com/kyverno/website/issues/1 -->
  - [x] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.

## Further Comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution
you did and what alternatives you considered, etc...
-->

JimBugwadia commented on charts/kyverno/templates/crds.yaml:
```diff
@@ -966,7 +967,7 @@ spec:
                 type: string
               webhookTimeoutSeconds:
-                description: WebhookTimeoutSeconds specifies the webhook timeout for this policy. After the timeout passes, the admission request will fail based on the failure policy. The default timeout is 3s, the value must be between 1 and 30 seconds. Default to 10 seconds.
+                description: WebhookTimeoutSeconds specifies the webhook timeout for this policy. After the timeout passes, the admission request will fail based on the failure policy. The default timeout is 10s, the value must be between 1 and 30 seconds.
```
Suggest rewording as:

> WebhookTimeoutSeconds specifies the maximum time in seconds allowed to apply this policy. After the configured time expires, the admission request may fail, or may simply ignore the policy results, based on the failure policy. The default timeout is 10s, the value must be between 1 and 30 seconds.

JimBugwadia commented on charts/kyverno/templates/crds.yaml:
```diff
@@ -3120,7 +3122,7 @@ spec:
                 type: string
               webhookTimeoutSeconds:
-                description: WebhookTimeoutSeconds specifies the webhook timeout for this policy. After the timeout passes, the admission request will fail based on the failure policy. The default timeout is 3s, the value must be between 1 and 30 seconds. Default to 10 seconds.
+                description: WebhookTimeoutSeconds specifies the webhook timeout for this policy. After the timeout passes, the admission request will fail based on the failure policy. The default timeout is 10s, the value must be between 1 and 30 seconds.
```
same as above...

JimBugwadia commented on pkg/webhookconfig/policy.go:
```diff
@@ -24,16 +24,19 @@ func (wrc *Register) contructPolicyValidatingWebhookConfig(caData []byte) *admre
+					APIVersions: []string{"v1"},
+				},
 				[]admregapi.OperationType{admregapi.Create, admregapi.Update},
+				admregapi.Ignore,
```
should this be set to `ignore` or `fail`?

realshuting replied:
No, k8s allows `Ignore` or `Fail`.
```
   failurePolicy	<string>
     FailurePolicy defines how unrecognized errors from the admission endpoint
     are handled - allowed values are Ignore or Fail. Defaults to Fail.
```

JimBugwadia commented on pkg/webhookconfig/policy.go:
```diff
@@ -73,16 +79,19 @@ func (wrc *Register) contructPolicyMutatingWebhookConfig(caData []byte) *admrega
+					APIVersions: []string{"v1"},
+				},
 				[]admregapi.OperationType{admregapi.Create, admregapi.Update},
+				admregapi.Ignore,
```
Same question....should this be set to ignore or fail?

JimBugwadia reviewed (commented):
Looks good! Some minor comments and questions.

JimBugwadia reviewed (changes requested):
lets update the API doc text, and then merge!

realshuting commented:
> lets update the API doc text, and then merge!

Are you referring this doc https://kyverno.io/docs/crds/? I'll update it with other doc changes.

realshuting commented:
The e2e test kept failing but it passed in my local env, see logs below. I'm going to merge this PR to avoid further conflicts. 
<details>
  <summary>make test-e2e-local</summary>

kubectl apply -f https://raw.githubusercontent.com/kyverno/kyverno/main/definitions/github/rbac.yaml
clusterrole.rbac.authorization.k8s.io/kyverno:userinfo unchanged
clusterrole.rbac.authorization.k8s.io/kyverno:customresources unchanged
clusterrole.rbac.authorization.k8s.io/kyverno:policycontroller unchanged
clusterrole.rbac.authorization.k8s.io/kyverno:generatecontroller unchanged
kubectl port-forward -n kyverno service/kyverno-svc-metrics  8000:8000 &
go test ./test/e2e/metrics -v
Unable to listen on port 8000: Listeners failed to create with the following errors: [unable to create listener: Error listen tcp4 127.0.0.1:8000: bind: address already in use unable to create listener: Error listen tcp6 [::1]:8000: bind: address already in use]
error: unable to listen on any of the requested ports: [{8000 8000}]
=== RUN   Test_MetricsServerAvailability
--- PASS: Test_MetricsServerAvailability (0.02s)
PASS
ok      github.com/kyverno/kyverno/test/e2e/metrics     (cached)
go test ./test/e2e/mutate -v
=== RUN   Test_Mutate_Sets
STEP: Test to mutate objects : test-mutate-with-context
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test-mutate
STEP: Creating Namespace 
STEP: 
Creating source ConfigMap in test-mutate
STEP: 
Creating Mutate ConfigMap Policy in 
STEP: 
Creating target ConfigMap in test-mutate
STEP: Verifying ConfigMap in the Namespace : test-mutate
STEP: configMap : {"apiVersion":"v1","data":{"data.yaml":"some: data\n"},"kind":"ConfigMap","metadata":{"annotations":{"policies.kyverno.io/last-applied-patches":"gen-role.mutate-policy.kyverno.io: added /metadata/labels/kyverno.key~1copy-me\n"},"creationTimestamp":"2021-10-05T06:30:52Z","labels":{"kyverno.key/copy-from":"source","kyverno.key/copy-me":"sample-value"},"managedFields":[{"apiVersion":"v1","fieldsType":"FieldsV1","fieldsV1":{"f:data":{".":{},"f:data.yaml":{}},"f:metadata":{"f:labels":{".":{},"f:kyverno.key/copy-from":{}}}},"manager":"mutate.test","operation":"Update","time":"2021-10-05T06:30:52Z"}],"name":"target","namespace":"test-mutate","resourceVersion":"74060","uid":"ef9cb1f1-52af-4f66-8cb1-b4ab1c1bc875"}}
STEP: Test test-mutate-with-context Completed 



STEP: Test to mutate objects : test-mutate-with-logic-in-context
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test-mutate
STEP: Creating Namespace 
STEP: 
Creating source ConfigMap in test-mutate
STEP: 
Creating Mutate ConfigMap Policy in 
STEP: 
Creating target ConfigMap in test-mutate
STEP: Verifying ConfigMap in the Namespace : test-mutate
STEP: configMap : {"apiVersion":"v1","data":{"data.yaml":"some: data\n"},"kind":"ConfigMap","metadata":{"annotations":{"policies.kyverno.io/last-applied-patches":"gen-role.mutate-policy.kyverno.io: added /metadata/labels/kyverno.key~1copy-me\n"},"creationTimestamp":"2021-10-05T06:31:03Z","labels":{"kyverno.key/copy-from":"source","kyverno.key/copy-me":"sample-value"},"managedFields":[{"apiVersion":"v1","fieldsType":"FieldsV1","fieldsV1":{"f:data":{".":{},"f:data.yaml":{}},"f:metadata":{"f:labels":{".":{},"f:kyverno.key/copy-from":{}}}},"manager":"mutate.test","operation":"Update","time":"2021-10-05T06:31:03Z"}],"name":"target","namespace":"test-mutate","resourceVersion":"74118","uid":"86599ce6-56bf-43d9-99cd-283aec952433"}}
STEP: Test test-mutate-with-logic-in-context Completed 



STEP: Test to mutate objects : test-mutate-with-context-label-selection
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test-mutate
STEP: Creating Namespace 
STEP: 
Creating source ConfigMap in test-mutate
STEP: 
Creating Mutate ConfigMap Policy in 
STEP: 
Creating target ConfigMap in test-mutate
STEP: Verifying ConfigMap in the Namespace : test-mutate
STEP: configMap : {"apiVersion":"v1","data":{"data.yaml":"some: data\n"},"kind":"ConfigMap","metadata":{"annotations":{"policies.kyverno.io/last-applied-patches":"gen-role.mutate-policy.kyverno.io: added /metadata/labels/kyverno.key~1copy-me\n"},"creationTimestamp":"2021-10-05T06:31:14Z","labels":{"kyverno.key/copy-from":"source","kyverno.key/copy-me":"sample-value"},"managedFields":[{"apiVersion":"v1","fieldsType":"FieldsV1","fieldsV1":{"f:data":{".":{},"f:data.yaml":{}},"f:metadata":{"f:labels":{".":{},"f:kyverno.key/copy-from":{}}}},"manager":"mutate.test","operation":"Update","time":"2021-10-05T06:31:14Z"}],"name":"target","namespace":"test-mutate","resourceVersion":"74168","uid":"64ea6545-987a-4095-a004-b9303d6ddb15"}}
STEP: Test test-mutate-with-context-label-selection Completed 



--- PASS: Test_Mutate_Sets (32.54s)
=== RUN   Test_Mutate
STEP: Mutation Test: checks that runAsNonRoot is added to security context and containers elements security context
STEP: Deleting Cluster Policies...
STEP: Deleting Resource...
STEP: Deleting Namespace...
STEP: Deleting Namespace: test-mutate...
STEP: Wait Till Deletion of Namespace...
STEP: Creating Namespace: ...
STEP: Wait Till Creation of Namespace...
STEP: Creating Policy...
STEP: Creating Resource...
STEP: Checking that resource is created...
STEP: Validating created resource with the expected pattern...
STEP: Deleting Cluster Policies...
STEP: Deleting Resource...
STEP: Deleting Namespace...
STEP: Wait Till Creation of Namespace...
STEP: Done
STEP: Mutation Test: checks that runAsNonRoot is added to security context and containers elements security context and initContainers elements security context
STEP: Deleting Cluster Policies...
STEP: Deleting Resource...
STEP: Deleting Namespace...
STEP: Deleting Namespace: test-mutate1...
STEP: Wait Till Deletion of Namespace...
STEP: Creating Namespace: ...
STEP: Wait Till Creation of Namespace...
STEP: Creating Policy...
STEP: Creating Resource...
STEP: Checking that resource is created...
STEP: Validating created resource with the expected pattern...
STEP: Deleting Cluster Policies...
STEP: Deleting Resource...
STEP: Deleting Namespace...
STEP: Wait Till Creation of Namespace...
STEP: Done
STEP: Mutation Test: checks that variables in the keys are working correctly
STEP: Deleting Cluster Policies...
STEP: Deleting Resource...
STEP: Deleting Namespace...
STEP: Deleting Namespace: test-mutate2...
STEP: Wait Till Deletion of Namespace...
STEP: Creating Namespace: ...
STEP: Wait Till Creation of Namespace...
STEP: Creating Policy...
STEP: Creating Resource...
STEP: Checking that resource is created...
STEP: Validating created resource with the expected pattern...
STEP: Deleting Cluster Policies...
STEP: Deleting Resource...
STEP: Deleting Namespace...
STEP: Wait Till Creation of Namespace...
STEP: Done
--- PASS: Test_Mutate (51.94s)
=== RUN   Test_Mutate_Ingress
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test-ingress
STEP: Creating mutate ClusterPolicy
STEP: Creating Namespace test-ingress
STEP: 

Start testing test-networking-v1-ingress
STEP: Creating Ingress networking.k8s.io/v1, Resource=ingresses in test-ingress
STEP: Verifying Ingress networking.k8s.io/v1, Resource=ingresses in the Namespace : test-ingress
STEP: Comparing patched field
STEP: 

Start testing test-networking-v1beta1-ingress
STEP: Creating Ingress networking.k8s.io/v1beta1, Resource=ingresses in test-ingress
W1004 23:32:16.021884   31327 warnings.go:70] networking.k8s.io/v1beta1 Ingress is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
STEP: Verifying Ingress networking.k8s.io/v1beta1, Resource=ingresses in the Namespace : test-ingress
W1004 23:32:16.025772   31327 warnings.go:70] networking.k8s.io/v1beta1 Ingress is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
STEP: Comparing patched field
STEP: 

Start testing test-extensions-v1beta1-ingress
STEP: Creating Ingress extensions/v1beta1, Resource=ingresses in test-ingress
W1004 23:32:16.039072   31327 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
STEP: Verifying Ingress extensions/v1beta1, Resource=ingresses in the Namespace : test-ingress
W1004 23:32:16.042509   31327 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
STEP: Comparing patched field
--- PASS: Test_Mutate_Ingress (3.13s)
PASS
ok      github.com/kyverno/kyverno/test/e2e/mutate      (cached)
go test ./test/e2e/generate -v
=== RUN   Test_ClusterRole_ClusterRoleBinding_Sets
STEP: Test to generate ClusterRole and ClusterRoleBinding : test-clusterrole-clusterrolebinding-without-clone
STEP: synchronize = false        clone = false
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test

STEP: Creating Generate Role Policy: gen-cluster-policy
Handling connection for 8000
STEP: Creating Namespace which triggers generate  

STEP: Verifying ClusterRole
STEP: Verifying ClusterRoleBinding
STEP: Test test-clusterrole-clusterrolebinding-without-clone Completed 



STEP: Test to generate ClusterRole and ClusterRoleBinding : test-clusterrole-clusterrolebinding-with-sync-without-clone
STEP: synchronize = true         clone = false
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test

STEP: Creating Generate Role Policy: gen-cluster-policy
STEP: Creating Namespace which triggers generate  

STEP: Verifying ClusterRole
STEP: Verifying ClusterRoleBinding
STEP: Test test-clusterrole-clusterrolebinding-with-sync-without-clone Completed 



STEP: Test to generate ClusterRole and ClusterRoleBinding : test-clusterrole-clusterrolebinding-with-sync-with-clone
STEP: synchronize = false        clone = true
STEP: Cleaning Cluster Policies
STEP: Clone = true, Deleting Source ClusterRole and ClusterRoleBinding
STEP: Deleting Namespace : test

STEP: Creating Generate Role Policy: gen-cluster-policy
STEP: Clone = true, Creating Cloner Resources in Namespace : 
STEP: Creating Namespace which triggers generate  

STEP: Verifying ClusterRole
STEP: Verifying ClusterRoleBinding
STEP: Test test-clusterrole-clusterrolebinding-with-sync-with-clone Completed 



--- PASS: Test_ClusterRole_ClusterRoleBinding_Sets (36.58s)
=== RUN   Test_Role_RoleBinding_Sets
STEP: Test to generate Role and RoleBinding : test-role-rolebinding-without-clone
STEP: synchronize = false        clone = false
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test
STEP: 
Creating Generate Role Policy in 
STEP: Creating Namespace which triggers generate 
STEP: Verifying Role in the Namespace : test
STEP: Verifying RoleBinding in the Namespace : test
STEP: Test test-role-rolebinding-without-clone Completed 



STEP: Test to generate Role and RoleBinding : test-role-rolebinding-withsync-without-clone
STEP: synchronize = true         clone = false
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test
STEP: 
Creating Generate Role Policy in 
STEP: Creating Namespace which triggers generate 
STEP: Verifying Role in the Namespace : test
STEP: Verifying RoleBinding in the Namespace : test
STEP: Test test-role-rolebinding-withsync-without-clone Completed 



STEP: Test to generate Role and RoleBinding : test-role-rolebinding-with-clone
STEP: synchronize = false        clone = true
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test
STEP: Clone = true, Deleting Source Role and RoleBinding from Clone Namespace : default
STEP: 
Creating Generate Role Policy in 
STEP: Clone = true, Creating Cloner Resources in Namespace : default
STEP: Creating Namespace which triggers generate 
STEP: Verifying Role in the Namespace : test
STEP: Verifying RoleBinding in the Namespace : test
STEP: Clone = true, Deleting Cloner Resources in Namespace : default
STEP: Test test-role-rolebinding-with-clone Completed 



--- PASS: Test_Role_RoleBinding_Sets (30.49s)
=== RUN   Test_Generate_NetworkPolicy
STEP: Test to generate NetworkPolicy : test-generate-policy-for-namespace-with-label
STEP: synchronize = true         clone = false
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test
STEP: Creating Generate NetworkPolicy Policy
STEP: Creating Namespace which triggers generate 
STEP: Verifying NetworkPolicy in the Namespace : test
STEP: Test test-generate-policy-for-namespace-with-label Completed 



--- PASS: Test_Generate_NetworkPolicy (10.12s)
=== RUN   Test_Generate_Namespace_Label_Actions
STEP: Test to generate NetworkPolicy : test-generate-policy-for-namespace-label-actions
STEP: synchronize = true         clone = false
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test
STEP: Creating Generate NetworkPolicy Policy
STEP: Creating Namespace which should not triggers generate policy 
STEP: Verifying NetworkPolicy in the Namespace : test
STEP: Updating Namespace which triggers generate policy 
STEP: Verifying NetworkPolicy in the updated Namespace : test
STEP: Update generate policy: add-networkpolicy
STEP: Updating Generate NetworkPolicy
STEP: Verifying updated NetworkPolicy in the Namespace : test
STEP: Test test-generate-policy-for-namespace-label-actions Completed 



--- PASS: Test_Generate_Namespace_Label_Actions (26.27s)
=== RUN   Test_Generate_Synchronize_Flag
STEP: Test to generate NetworkPolicy : test-generate-policy-for-namespace-with-label
STEP: synchronize = true         clone = false
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test
STEP: Creating Generate NetworkPolicy Policy
STEP: Creating Namespace which triggers generate 
STEP: Verifying NetworkPolicy in the Namespace : test
STEP: Deleting NetworkPolicy allow-dns in the Namespace : test
STEP: Checking NetworkPolicy allow-dns in the Namespace : test
STEP: Update synchronize to true in generate policy: add-networkpolicy
STEP: Verify the label in the updated network policy: allow-dns
STEP: Deleting NetworkPolicy allow-dns in the Namespace : test
STEP: Checking NetworkPolicy allow-dns in the Namespace : test
STEP: Test test-generate-policy-for-namespace-with-label Completed 



--- PASS: Test_Generate_Synchronize_Flag (12.24s)
=== RUN   Test_Source_Resource_Update_Replication
STEP: Test to check replication of clone source resource: test-clone-source-resource-update-replication
STEP: synchronize = true         clone = true
STEP: Cleaning Cluster Policies
STEP: Deleting Namespace : test
STEP: Clone = true, Deleting Source Resource from Clone Namespace : default
STEP: Clone = true, Creating Cloner Resources in Namespace : default
STEP: 
Creating Generate Policy in 
STEP: Creating Namespace which triggers generate 
STEP: Verifying Configmap in the Namespace : test
STEP: Updating Source Resource(Configmap) in Clone Namespace : default
STEP: Verifying Configmap Data Replication in the Namespace : test
STEP: Updating Generated ConfigMap in Resource Namespace : test
STEP: Verifying Configmap Data in the Namespace : test
STEP: Clone = true, Deleting Cloner Resources in Namespace : default
STEP: Test test-clone-source-resource-update-replication Completed 



--- PASS: Test_Source_Resource_Update_Replication (13.23s)
=== RUN   Test_Generate_Policy_Deletion_for_Clone
STEP: Test to check kyverno flow when generate policy is deleted: test-clone-source-resource-update-replication
STEP: synchronize = true         clone = true
STEP: Cleaning Cluster Policies
STEP: Clone = true, Deleting Source Resource from Clone Namespace : default
STEP: Deleting Namespace : test
STEP: Clone = true, Creating Cloner Resources in Namespace : default
STEP: 
Creating Generate Policy in 
STEP: Creating Namespace which triggers generate 
STEP: Verifying Configmap in the Namespace : test
STEP: Delete the generate policy : generate-policy
STEP: Checking the generated resource (Configmap) in namespace : test
STEP: Updating Source Resource(Configmap) in Clone Namespace : default
STEP: Verifying Configmap Data is Not Replicated in Generated Resource
STEP: Delete the clone source resource: default/game-demo
STEP: Checking the generated resource (Configmap) in namespace : test
STEP: Clone = true, Deleting Cloner Resources in Namespace : default
STEP: Test test-clone-source-resource-update-replication Completed 



--- PASS: Test_Generate_Policy_Deletion_for_Clone (39.92s)
PASS
ok      github.com/kyverno/kyverno/test/e2e/generate    169.130s
</details>

Suspecting it relates to https://github.com/kyverno/kyverno/issues/2208, cc @NoSkillGirl .
